app/javascript/components/student/solution-summary/MentoringComboButton.tsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import { usePanel } from '../../../hooks/use-panel'
+import { MentoringDropdown } from '../MentoringDropdown'
+import { GraphicalIcon } from '../../common'
+import { MentorDiscussion } from '../../types'
+
+type Links = {
+  requestMentoring: string
+  shareMentoring: string
+  pendingMentorRequest: string
+  inProgressDiscussion?: string
+}
+
+export const MentoringComboButton = ({
+  hasMentorDiscussionInProgress,
+  hasMentorRequestPending,
+  discussions,
+  className = '',
+  links,
+}: {
+  hasMentorDiscussionInProgress: boolean
+  hasMentorRequestPending: boolean
+  discussions: readonly MentorDiscussion[]
+  className?: string
+  links: Links
+}): JSX.Element => {
+  const {
+    open,
+    setOpen,
+    setButtonElement,
+    setPanelElement,
+    styles,
+    attributes,
+  } = usePanel({
+    placement: 'bottom-end',
+    modifiers: [
+      {
+        name: 'offset',
+        options: {
+          offset: [0, 14],
+        },
+      },
+    ],
+  })
+
+  const classNames = ['c-combo-button', className]
+
+  return (
+    /* TODO: Extract into a common component in the future */
```

Comment:
Can this now be removed, or is this about certain child parts of the combo button?